pkg/providers/v1/aws_sdk.go

```diff
@@ -211,18 +211,17 @@ func (p *awsSDKProvider) Metadata() (config.EC2Metadata, error) {
 	p.addAPILoggingHandlers(&client.Handlers)
 
 	identity, err := client.GetInstanceIdentityDocument()
-	if err != nil {
-		return nil, fmt.Errorf("unable to get instance identity document: %v", err)
```

Comment:
Can you add a log line for this error?

Reply:
Thanks for the review!

The line would be logged for every invocation of the function and it would basically be a log line that says  "I wanted to log something useful but the call failed so I'm logging the fact that I couldn't log something useful".

How often is this function called?

Reply:
Calls to IMDS can fail for reasons other than `AWS_EC2_METADATA_DISABLED`, so I think its worth logging. It's only called once in an `init()`: https://github.com/kubernetes/cloud-provider-aws/blob/d7e05d57709cd46297490b51ce0dd11a54dbea35/pkg/providers/v1/aws.go#L473